crates/rpc/lean/src/handlers/block.rs

```diff
@@ -23,20 +24,35 @@ pub async fn get_block(
 pub async fn get_block_by_id(
     block_id: ID,
     lean_chain: Data<LeanChainReader>,
-) -> Result<Option<Block>, ApiError> {
-    // Obtain read guard first from the reader.
-    let lean_chain = lean_chain.read().await;
+) -> Result<Option<SignedBlock>, ApiError> {
```

Comment:
```suggestion
) -> Result<Option<Block>, ApiError> {
```

Let's just return `Block` instead of `SignedBlock`